solutions/usaco-596.mdx

```diff
@@ -93,7 +93,66 @@ void ff(int i, int j)
 	}
 }
 ```
+## Alternate Faster Solution - Time Complexity $\mathcal{O}(N\log N)$
 
+It is possible to make the following 2 observations:
+
+1. For each contiguous closed region, there must be 1 gate. In a minimum state, each region has exactly 1 gate.
+2. Each time FJ comes across a node he has visited along an edge he has not crossed in either direction, he creates exactly 1 new region. In other words, FJ creates exactly 1 new contiguous close region if:
```

Comment:
maybe clarify something like "if all the following conditions are satisfied"

Reply:
Done

Reply:
Done